rational/rational.go

```diff
@@ -0,0 +1,203 @@
+func New(Numerator int64, Denominator ...int64) Rat {
+	switch len(Denominator) {
+	case 0:
+		return Rat{big.NewRat(Numerator, 1)}
+	case 1:
+		return Rat{big.NewRat(Numerator, Denominator[0])}
+	default:
+		panic("improper use of New, can only have one denominator")
+	}
+}
+
+//NewFromDecimal - create a rational from decimal string or integer string
+func NewFromDecimal(decimalStr string) (f Rat, err error) {
+
+	// first extract any negative symbol
+	neg := false
+	if string(decimalStr[0]) == "-" {
+		neg = true
+		decimalStr = decimalStr[1:]
+	}
+
+	str := strings.Split(decimalStr, ".")
+
+	var numStr string
+	var denom int64 = 1
+	switch len(str) {
+	case 1:
+		if len(str[0]) == 0 {
+			return f, fmt.Errorf("not a decimal string")
+		}
+		numStr = str[0]
+	case 2:
+		if len(str[0]) == 0 || len(str[1]) == 0 {
+			return f, fmt.Errorf("not a decimal string")
+		}
+		numStr = str[0] + str[1]
+		len := int64(len(str[1]))
+		denom = new(big.Int).Exp(big.NewInt(10), big.NewInt(len), nil).Int64()
+	default:
+		return f, fmt.Errorf("not a decimal string")
+	}
+
+	num, err := strconv.Atoi(numStr)
+	if err != nil {
+		return f, err
+	}
+
+	if neg {
+		num *= -1
+	}
+
+	return Rat{big.NewRat(int64(num), denom)}, nil
+}
+
+//nolint
+func (r Rat) GetRat() *big.Rat  { return r.Rat }                                     // GetRat - get big.Rat
+func (r Rat) Num() int64        { return r.Rat.Num().Int64() }                       // Num - return the numerator
+func (r Rat) Denom() int64      { return r.Rat.Denom().Int64() }                     // Denom  - return the denominator
+func (r Rat) IsZero() bool      { return r.Num() == 0 }                              // IsZero - Is the Rat equal to zero
+func (r Rat) Equal(r2 Rat) bool { return r.Rat.Cmp(r2.GetRat()) == 0 }               // Equal - rationals are equal
+func (r Rat) GT(r2 Rat) bool    { return r.Rat.Cmp(r2.GetRat()) == 1 }               // GT - greater than
+func (r Rat) LT(r2 Rat) bool    { return r.Rat.Cmp(r2.GetRat()) == -1 }              // LT - less than
+func (r Rat) Inv() Rat          { return Rat{new(big.Rat).Inv(r.Rat)} }              // Inv - inverse
+func (r Rat) Mul(r2 Rat) Rat    { return Rat{new(big.Rat).Mul(r.Rat, r2.GetRat())} } // Mul - multiplication
+func (r Rat) Quo(r2 Rat) Rat    { return Rat{new(big.Rat).Quo(r.Rat, r2.GetRat())} } // Quo - quotient
+func (r Rat) Add(r2 Rat) Rat    { return Rat{new(big.Rat).Add(r.Rat, r2.GetRat())} } // Add - addition
+func (r Rat) Sub(r2 Rat) Rat    { return Rat{new(big.Rat).Sub(r.Rat, r2.GetRat())} } // Sub - subtraction
+
+var zero = big.NewInt(0)
+var one = big.NewInt(1)
+var two = big.NewInt(2)
+var five = big.NewInt(5)
+var nFive = big.NewInt(-5)
+var ten = big.NewInt(10)
+
+// EvaluateBig - evaluate the rational using bankers rounding
+func (r Rat) EvaluateBig() *big.Int {
+
+	num := r.Rat.Num()
+	denom := r.Rat.Denom()
+
+	d, rem := new(big.Int), new(big.Int)
+	d.QuoRem(num, denom, rem) // always drops the decimal
```

Comment:
What does this comment mean? I figured that:
quotient    = num/denom  --> an int
remainder = num%denom --> an int
so perhaps this comment is not relevant?

Reply:
yup comment obsolete removing. Thanks